crates/driver/src/domain/eth/gas.rs

```diff
@@ -72,6 +72,22 @@ impl GasPrice {
     }
 }
 
+/// Implements multiplication of a gas price by a floating point number.
+/// This is equivalent to multiplying the `tip` and adjusting the max fee to
+/// accommodate the new tip.
+impl std::ops::Mul<f64> for GasPrice {
+    type Output = Self;
+
+    fn mul(self, rhs: f64) -> Self::Output {
+        let new_tip = self.tip.mul_ceil(rhs).into();
+        Self {
+            max: (self.max + new_tip).saturating_sub(self.tip),
```

Comment:
Old code bumps both max and tip by 12.5% and if i remember correctly this is a requirement from most of the clients to replace an existing transaction in mempool.

Reply:
You are right ([here](https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.TxPool/Comparison/CompareReplacedTxByFee.cs#L43-L55) is the code to check). How annoying (if we start off with a very high max for simplicity, gas requirements of our solver may be pretty high for cancellations.